pkg/catalog/manager/sync.go

```diff
@@ -2,12 +2,90 @@ package manager
 
 import (
 	"github.com/rancher/types/apis/management.cattle.io/v3"
+	"github.com/rancher/types/client/management/v3"
 	"github.com/sirupsen/logrus"
 	kerrors "k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/tools/cache"
 )
 
+func (m *Manager) PrjCatalogSync(key string, obj *v3.ProjectCatalog) error {
+	m.commonSync(key, &obj.Catalog)
+	cmt := &CatalogManagerType{
+		catalog:        &obj.Catalog,
+		projectCatalog: obj,
+		catalogType:    client.ProjectCatalogType,
+	}
+
+	ns, name, err := cache.SplitMetaNamespaceKey(key)
+	if err != nil {
+		return err
+	}
+
+	projectCatalog, err := m.projectCatalogClient.GetNamespaced(ns, name, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+
+	repoPath, commit, err := m.prepareRepoPath(obj.Catalog)
+	if err != nil {
+		v3.CatalogConditionRefreshed.False(projectCatalog)
+		v3.CatalogConditionRefreshed.ReasonAndMessageFromError(projectCatalog, err)
+		m.projectCatalogClient.Update(projectCatalog)
+		return err
+	}
+
+	if commit == projectCatalog.Status.Commit {
+		logrus.Debugf("Catalog %s is already up to date", projectCatalog.Name)
+		if v3.CatalogConditionRefreshed.IsUnknown(projectCatalog) {
+			v3.CatalogConditionRefreshed.True(projectCatalog)
+			v3.CatalogConditionRefreshed.Reason(projectCatalog, "")
+			m.projectCatalogClient.Update(projectCatalog)
+		}
+		return nil
+	}
+
+	logrus.Infof("Updating catalog %s", projectCatalog.Name)
+	return m.traverseAndUpdate(repoPath, commit, cmt)
+}
+
 func (m *Manager) Sync(key string, obj *v3.Catalog) error {
+	m.commonSync(key, obj)
```

Comment:
First sync method should make a check to see if 'obj' is nil and return if it is.